src/Microsoft.Azure.SignalR.AspNet/EndpointProvider/ServiceEndpointProvider.cs

```diff
@@ -95,9 +98,10 @@ public string GetClientEndpoint(string hubName = null, string originalPath = nul
 
         public string GetServerEndpoint(string hubName)
         {
+            var prefixedHubName = string.IsNullOrEmpty(_appName) ? hubName.ToLower() : $"{_appName.ToLower()}_{hubName.ToLower()}";
             return _port.HasValue ?
-                $"{_endpoint}:{_port}/{ServerPath}/?hub={hubName.ToLower()}" :
-                $"{_endpoint}/{ServerPath}/?hub={hubName.ToLower()}";
+                $"{_endpoint}:{_port}/{ServerPath}/?hub={prefixedHubName.ToLower()}" :
+                $"{_endpoint}/{ServerPath}/?hub={prefixedHubName.ToLower()}";
```

Comment:
no need to `ToLower` as `prefixedHubName` is already lowered

Reply:
fixed